nibabel/funcs.py

```diff
@@ -120,19 +120,13 @@ def concat_images(images, check_affines=True, axis=None):
     klass = img0.__class__
     shape0 = img0.shape
     n_dim = len(shape0)
-    if axis is None:
-        # collect images in output array for efficiency
-        out_shape = (n_imgs, ) + shape0
```

Comment:
Ah no - this was different deliberately in order to save memory.  With this path, there are not large temporary arrays for the axis == None case, we directly fill the output array.